### PR TITLE
[serverless] Use simple format for serialised checkpoints

### PR DESCRIPTION
**Pull request by AlCutter**

### CODE REVIEW

File: serverless/api/state.go [hunk 1]
```diff
@@ -15,6 +15,17 @@
...
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+// CheckpointHeader is the first line of a marshaled log checkpoint.
+const CheckpointHeader = "Log Checkpoint v0"
```
Comment (mhutchinson) [inline serverless/api/state.go:27]: Stick v0 in the const name (or make it private/lowercase)?

File: serverless/api/state.go [hunk 2]
```diff
@@ -24,6 +35,44 @@ type LogState struct {
...
+		return errors.New("invalid checkpoint - incorrect header")
+	}
+	size, err := strconv.ParseUint(l[1], 10, 64)
+	if err != nil {
+		return fmt.Errorf("invalid checkpoint - size invalid: %w", err)
+	}
+	rh, err := base64.StdEncoding.DecodeString(l[2])
+	if err != nil {
+		return fmt.Errorf("invalid checkpoint - invalid roothash: %w", err)
+
```
Comment (mhutchinson) [inline serverless/api/state.go:70]:
```suggestion
```

File: serverless/api/state.go [hunk 3]
```diff
@@ -24,6 +35,44 @@ type LogState struct {
...
+	}
+	s.Size, s.RootHash = size, rh
```
Comment (mhutchinson) [inline serverless/api/state.go:72]: Consider checking the size of `rh`.  The doc for `RootHash` says it's 256 bits so may as well be paranoid on inputs. Though I realize this totally makes your tests less fun. So YMMV :-)

**Review by mhutchinson — APPROVED** [review]
nit: also change PR description to s/add a/changed to/ ; this changes the format used.

File: serverless/api/state.go (hunk 1 above)
Comment (AlCutter) [reply serverless/api/state.go:27]: done

File: serverless/api/state.go (hunk 2 above)
Comment (AlCutter) [reply serverless/api/state.go:70]: done

File: serverless/api/state.go (hunk 3 above)
Comment (AlCutter) [reply serverless/api/state.go:72]: I think that's probably the layer above (unless I make the struct have a `[32]byte` for `RootHash`?), maybe I'll update the comment instead :)

**Review by AlCutter — COMMENTED** [review]
> nit: also change PR description to s/add a/changed to/ ; this changes the format used.

Good call, done

**Comment by AlCutter** [comment]
(rebased)

**Pull request merged**